viewer/viewer.js

```diff
@@ -3580,6 +3585,204 @@ app.post('/updateUser/:userId', checkToken, function(req, res) {
   });
 });
 
+app.post('/uploadFragment', checkWebEnabled, fragmentUpload, function(req, res) {
+  Db.getUser(req.user.userId, function(err, user) {
```

Comment:
all these functions that call error need a 

  function error(text) {
    return res.send(JSON.stringify({success: false, text: text}));
  }